src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -714,14 +716,17 @@ public override ClaimsPrincipal ValidateToken(string token, TokenValidationParam
                 var decryptedJwt = DecryptToken(jwtToken, validationParameters);
                 var innerToken = ValidateSignature(decryptedJwt, validationParameters);
                 jwtToken.InnerToken = innerToken;
-                validatedToken = jwtToken;
-                return ValidateTokenPayload(innerToken, validationParameters);
+                signatureValidatedToken = jwtToken;
+                claimsPrincipal = ValidateTokenPayload(innerToken, validationParameters);
+                validatedToken = signatureValidatedToken;
```

Comment:
You could set validatedToken and return claimsPrincipal after the if/else statement.